agent/vendor/github.com/aws/amazon-ecs-agent/ecs-agent/acs/session/heartbeat_responder.go

```diff
@@ -11,40 +11,64 @@
 // express or implied. See the License for the specific language governing
 // permissions and limitations under the License.
 
-package handler
+package session
 
 import (
 	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
 	"github.com/aws/amazon-ecs-agent/ecs-agent/doctor"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger/field"
 	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/cihub/seelog"
 )
 
-func HeartbeatHandlerFunc(acsClient wsclient.ClientServer, doctor *doctor.Doctor) func(message *ecsacs.HeartbeatMessage) {
-	return func(message *ecsacs.HeartbeatMessage) {
-		handleSingleHeartbeatMessage(acsClient, doctor, message)
+// heartbeatResponder implements the wsclient.RequestResponder interface for responding
+// to ecsacs.HeartbeatMessage type.
+type heartbeatResponder struct {
+	doctor  *doctor.Doctor
+	respond wsclient.RespondFunc
+}
+
+// NewHeartbeatResponder returns an instance of the heartbeatResponder struct.
+func NewHeartbeatResponder(doctor *doctor.Doctor) *heartbeatResponder {
+	return &heartbeatResponder{
+		doctor: doctor,
 	}
 }
 
+func (*heartbeatResponder) Name() string {
+	return "heartbeat message responder"
+}
+
+func (r *heartbeatResponder) RegisterResponder(respond wsclient.RespondFunc) {
+	r.respond = respond
+}
+
+func (r *heartbeatResponder) HandlerFunc() wsclient.RequestHandler {
+	return r.processHeartbeatMessage
+}
+
 // To handle a Heartbeat Message the doctor health checks need to be run and
 // an ACK needs to be sent back to ACS.
 
 // This function is meant to be called from the ACS dispatcher and as such
 // should not block in any way to prevent starvation of the message handler
```

Comment:
Can we modify this comment block to follow the standard Go doc commentary for functions?

ie. `processHeartbeatMessage [does xyz]`